src/crypto/tls/ech_config.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2020 Cloudflare, Inc. All rights reserved. Use of this source code
+// is governed by a BSD-style license that can be found in the LICENSE file.
+
+package tls
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/cisco/go-hpke"
+
+	"golang.org/x/crypto/cryptobyte"
+)
+
+// ECHConfig represents an ECH configuration.
+type ECHConfig struct {
```

Comment:
I had used go-tls-syntax originally but @Lekensteyn asked that I replace it with cryptobyte, the reasoning being that it adds one less dependency. I think he's right.

Reply:
Let's not change this now, but: I really don't like that reasoning, *especially* since this is parsing code. We've just duplicated well-tested code for no reason. Libraries are meant to be re-used.

Reply:
The Go standard library already uses cryptobyte which is already more readable and less error-prone than manually bitshifting stuff and maintaining byte offsets as was done before. If this is ever going to be upstreamed, adding new dependencies such as go-tls-syntax is going to be make that process harder. And I don't think the advantages of go-tls-syntax are really that big, it's not solving a major problem.